src/core/components/model.jsx

```diff
@@ -3,6 +3,8 @@ import ImmutablePureComponent from "react-immutable-pure-component"
 import ImPropTypes from "react-immutable-proptypes"
 import PropTypes from "prop-types"
 
+export let decodeRefName = s => decodeURIComponent(s.replace(/~1/, "/").replace(/~0/, "~"))
```

Comment:
Again, thanks for contributing!

**1.** [decodeURIComponent](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/decodeURIComponent) can throw [URIError](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/URIError) in certain cases like for example: `c%d`. JSON Pointer specification unfortunately allow to use encoded strings like  `c%d`, which means we have to compensate for that cornser-case. More info in [RFC6901, section 5](https://www.rfc-editor.org/rfc/rfc6901#section-5).

**2.** The regex pattern you used only replaces first occurrence of encoded JSON Pointer symbols and ignore the rest.

Here is a suggestion for you function that addressed both the issues mentioned above:

```suggestion
export let decodeRefName = uri => {
  const unescaped = uri.replace(/~1/g, "/").replace(/~0/g, "~")
  
  try {
    return decodeURIComponent(unescaped)
  } catch {
    return unescaped
  }
}
```

PS: can I you pleas also provide more tests coverage for the cases that my suggestion tries to fix? Thanks!